### ISSUE TITLE
deferred error : (text-read-only)

### ISSUE THREAD

**Issue by fpopineau** [myuhe/org-gcal.el#28]
Hi,
This is the error I get when calling org-gcal-sync. However, both my calendars seems to be sync'ed.
I suspect this error might come from an auxiliary file.
I have tried to set:
(setq deferred:debug-on-signal nil)
but this time I hit various errors, none of which seems to prevent the downloading of the calendars.
It is a bit scary though. Any idea on how to debug this?

**Comment by fpopineau** [comment]
I found the reason for the (text-read-only) : when downloading the icon file, the png buffer is switching automatically to an image mode which sets the buffer read-only. Calling (fundamental-mode) before (insert ...) fixes it.

**Comment by zbeekman** [comment]
Hi I’m wondering if you can post a patch for this. I too have this error and am not skilled enough to fix it from your description.

**Comment by fpopineau** [comment]
Replace the function org-gcal--notify by this one :

(defun org-gcal--notify (title mes)
  (let ((file (expand-file-name (concat  (file-name-directory
(locate-library "org-gcal")) org-gcal-logo)))
        (mes mes)
        (title title))
    (if (eq system-type 'gnu/linux)
        (progn
          (if (not (file-exists-p file))
              (deferred:$
                (deferred:url-retrieve (concat "
https://raw.githubusercontent.com/myuhe/org-gcal.el/master/" org-gcal-logo))
                (deferred:nextc it
                  (lambda (buf)
                    (with-current-buffer buf
                      (let ((tmp (substring (buffer-string) (+
(string-match "\n\n" (buffer-string)) 2))))
                        (erase-buffer)
                        (fundamental-mode)
                        (insert tmp)
                        (write-file file)))
                    (kill-buffer buf)))))
          (alert mes :title title :icon file))
      (alert mes :title title))
    ))

Cheers,

2015-05-20 21:49 GMT+02:00 Izaak Beekman <notifications@github.com>:

**Comment by zbeekman** [comment]
Thanks, it seems like this fixed the issue… I might make a PR for @myuhe if I have a sec later

**Comment by maniux** [comment]
Does the author notice this?

**Comment by maniux** [comment]
@myuhe

**Comment by dakrone** [comment]
I am still seeing this issue with the latest MELPA install :(

**Comment by erikmd** [comment]
I assume that you are using MELPA, not MELPA Stable, correct? (the version of `org-gcal.el` in MELPA Stable being outdated)
Could you provide a backtrace of the error, by evaluating `(setq debug-on-error t)` beforehand?
And what is your OS and version of emacs / org-mode ?

**Comment by dakrone** [comment]
@erikmd I tried with both the MELPA stable and MELPA versions (same error).

Here's the backtrace (I took out the access_token and key):

```
Debugger entered--Lisp error: (error "Keyword argument :data not one of (:error-thrown)")
  signal(error ("Keyword argument :data not one of (:error-thrown)"))
  error("Keyword argument %s not one of (:error-thrown)" :data)
  #[128 "\300\301\"A@\211\2031
  apply(#[128 "\300\301\"A@\211\2031
  apply(apply #[128 "\300\301\"A@\211\2031
  request--safe-apply(#[128 "\300\301\"A@\211\2031
  request--callback(#<killed buffer> :type "GET" :params ((access_token . "...") (key . "...") (singleEvents . "True") (orderBy . "startTime") (timeMin . "2015-10-12T15:02:39Z") (timeMax . "2016-01-10T15:02:39Z") ("grant_type" . "authorization_code")) :parser org-gcal--json-read :error #[128 "\300\301\"A@\211\2031
```

Version details:

```
OS: Fedora 22
Emacs: GNU Emacs 25.0.50.2 (x86_64-unknown-linux-gnu, GTK+ Version 3.16.7) of 2015-11-10
Org: Org-mode version 8.3.2 (8.3.2-32-g73a5c2-elpa @ /home/hinmanm/.emacs.d/elpa/org-20151109/)
```

**Comment by erikmd** [comment]
Hi @dakrone,

Thanks for the more detailed info, and sorry for the delay.

Actually I am unable to reproduce your error, and I am unsure that this deals with 
this (currently closed) issue #28, as there is no occurrence of (text-read-only) in your backtrace.
(And if it were the same issue, only the original poster and the repo owner could reopen the issue.)

If you are still experiencing an issue, I can suggest the following:
- remove byte-compiled files `*.elc` (e.g., by hand) as they typically "pollute" the backtrace,
- replace `org-gcal.el` with the latest version on my testing branch https://github.com/erikmd/org-gcal.el/blob/testing/org-gcal.el (which incorporates the PR #45)
- restart Emacs and evaluate `(setq debug-on-error t deferred:debug-on-signal t)`
- run `(org-gcal-fetch)`
- and open a new issue mentioning the new backtrace and/or the potential errors that may appear in the `*Messages*` buffer (as well as your config).

Hoping that this helps.
